src/pages/workspace/reimburse/WorkspaceRateAndUnitPage.js

```diff
@@ -0,0 +1,170 @@
+        const numValue = this.getNumericValue(value);
+        if (Number.isNaN(numValue)) {
+            return '';
+        }
+        return numValue.toString().replace('.', this.props.toLocaleDigit('.')).substring(0, value.length);
+    }
+
+    getNumericValue(value) {
+        const numValue = parseFloat(value.toString().replace(',', '.'));
+        if (Number.isNaN(numValue)) {
+            return NaN;
+        }
+        return numValue.toFixed(3);
+    }
+
+    saveUnitAndRate(unit, rate) {
+        const distanceCustomUnit = _.find(lodashGet(this.props, 'policy.customUnits', {}), u => u.name === 'Distance');
+        if (!distanceCustomUnit) {
+            return;
+        }
+
+        const rateNumValue = this.getNumericValue(rate);
+        const currentCustomUnitRate = lodashGet(distanceCustomUnit, ['rates', this.state.unitRateID], {});
+
+        const newCustomUnit = {
+            customUnitID: this.state.unitID,
+            name: this.state.unitName,
+            attributes: {unit},
+            rates: {
+                ...currentCustomUnitRate,
+                rate: rateNumValue * CONST.POLICY.CUSTOM_UNIT_RATE_BASE_OFFSET,
+            },
+        };
+        Policy.updateWorkspaceCustomUnitAndRate(this.props.policy.id, distanceCustomUnit, newCustomUnit, this.props.policy.lastModified);
+    }
+
+    submit(values) {
+        this.saveUnitAndRate(values.unit, values.rate);
+        Keyboard.dismiss();
+        Navigation.navigate(ROUTES.getWorkspaceReimburseRoute(this.props.policy.id));
+    }
+
+    validate(values) {
+        const errors = {};
+        const decimalNumberRegex = new RegExp(CONST.REGEX.DECIMAL_NUMBER);
+        if (!decimalNumberRegex.test(values.rate)) {
+            errors.rate = this.props.translate('workspace.reimburse.invalidRateError');
+        }
+
+        return errors;
+    }
+
+    render() {
+        return (
+            <WorkspacePageWithSections
+                shouldUseScrollView
+                headerText={this.props.translate('workspace.reimburse.trackDistance')}
+                route={this.props.route}
+                guidesCallTaskID={CONST.GUIDES_CALL_TASK_IDS.WORKSPACE_REIMBURSE}
+                shouldSkipVBBACall
+                backButtonRoute={ROUTES.getWorkspaceReimburseRoute(this.props.policy.id)}
+            >
+                {() => (
+                    <Form
+                        formID={ONYXKEYS.FORMS.WORKSPACE_RATE_AND_UNIT_FORM}
+                        submitButtonText={this.props.translate('workspace.editor.save')}
+                        style={[styles.mh5, styles.flexGrow1]}
+                        scrollContextEnabled
+                        validate={this.validate}
+                        onSubmit={this.submit}
+                        enabledWhenOffline
+                    >
+                        <OfflineWithFeedback
+                            pendingAction={lodashGet(this.props.policy, 'pendingFields.generalSettings')}
```

Comment:
I changed it but I am not sure if it's good now, I don't know how to test it.

Reply:
There is a checklist item for exactly this 
>  I turned off my network connection and tested it while offline to ensure it matches the expected behavior

It should behave the same as workspace settings page when there is a pending action offline 

https://github.com/Expensify/App/assets/9059945/804a7335-74e8-4851-bf5c-8b2591d7b550

The same thing with errors, you can use reproduction steps from https://github.com/Expensify/App/issues/18447 to see how errors are handled by workspace settings page

Reply:
@eVoloshchak For me this is already working as expected for pending action offline:

https://github.com/Expensify/App/assets/19537677/3b073299-a981-4250-ae79-3986b9b1bd74



 For errors I don't see what errors could arrived from the server that are not already handled by the validate method bu if it was working before it should still be working because I used the same logic.